### ISSUE TITLE
Not working on virtual filename generated by ESLint preprocessor

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#3174]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

<pre>
  <code>
```TypeScript
export type Value = 1 | 2 | 3 | 4 | 5
```
  </code>
</pre>

![image](https://user-images.githubusercontent.com/8336744/110903256-1e0a3e00-8342-11eb-90c3-b6cf5b92aef0.png)

**Expected Result**

Work like normal ts files

**Actual Result**

```log
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: test/fixtures/code-block.md/2.ts.
The file must be included in at least one of the projects provided
```

**Additional Info**

related to https://github.com/eslint/eslint/issues/11989 and https://github.com/prettier/eslint-plugin-prettier/issues/393

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.16.1` |
| `TypeScript`                | `4.2.3` |
| `ESLint`                    | `7.21.0` |
| `node`                      | `12.21.0` |

**Comment by bradzacher** [comment]
Unfortunately we cannot support this usecase.
The TypeScript APIs we use require files to exist on the disk.

The "fake" files created by pre-processors obviously are not on disk, so they cannot be included in the TypeScript program.
